scripts/actions/compressed-size/main.js

```diff
@@ -0,0 +1,338 @@
+/* eslint-disable */
+
+import path from "path";
+import fs from "fs";
+import os from "os";
+import assert from "assert";
+import { Duplex } from "stream";
+
+import shell from "shelljs";
+import compressing from "compressing";
+import { startGroup, endGroup } from "@actions/core";
+import { context, getOctokit } from "@actions/github";
+import SizePlugin from "size-plugin-core";
+
+import { diffTable, toBool } from "./utils.js";
+
+let token = process.env.token;
+
+let owner, repo, ref, workflow, pull_number;
+
+({ ref, workflow } = context);
+({ owner, repo } = context.repo);
+pull_number = context.issue.number;
+
+const octokit = getOctokit(token);
+
+async function getArtifact(branch, actionName) {
+  assert(workflow !== branch, "Cannot use `getArtifact` in target workflow");
+
+  let i = 0;
+  while (true) {
+    i++;
+    try {
+      let runs = (
+        await octokit.actions.listWorkflowRunsForRepo({
+          owner,
+          repo,
+          branch: branch,
+        })
+      ).data.workflow_runs;
+
+      const run = runs.find(v => v.name === actionName);
+
+      if (run.status === "in_progress") {
+        throw new Error("Target workflow is still in progress");
+      }
+
+      //console.log("WorkflowRun", run);
+
+      const artifact = (
+        await octokit.actions.listWorkflowRunArtifacts({
+          owner,
+          repo,
+          run_id: run.id,
+        })
+      ).data.artifacts[0];
+
+      assert(artifact.name === "verdaccio-workspace");
+
+      //console.log("Artifact", artifact);
+
+      const buf = Buffer.from(
+        new Uint8Array(
+          (
+            await octokit.actions.downloadArtifact({
+              owner,
+              repo,
+              artifact_id: artifact.id,
+              archive_format: "zip",
+            })
+          ).data
+        )
+      );
+
+      assert(buf.length > 0);
+
+      return buf;
+    } catch (error) {
+      // 30 minutes
+      if (i > 180) {
+        throw error;
+      }
+      console.log("Retry to fetch artifact", i);
+      await new Promise(resolve => setTimeout(resolve, 10000));
```

Comment:
It has to wait for the `E2E tests` of the entire main branch to complete, otherwise the artifact cannot be downloaded.